enterprise/internal/insights/background/queryrunner/work_handler.go

```
@@ -119,7 +124,30 @@ func (r *workHandler) Handle(ctx context.Context, logger log.Logger, record *Job
 
 	recordings, err := executableHandler(ctx, &job.SearchJob, series, recordTime)
 	if err != nil {
+		if !r.baseWorkerStore.WillRetry(job) && isGlobal && job.PersistMode == string(store.RecordMode) {
+			reason := TranslateIncompleteReasons(err)
+			logger.Debug("insights recording global query timeout",
+				log.Int("seriesId", series.ID), log.String("seriesUniqueId", series.SeriesID),
+				log.Error(err),
+				log.String("reason", string(reason)))
+
+			if err := r.insightsStore.AddIncompleteDatapoint(ctx, store.AddIncompleteDatapointInput{
+				SeriesID: series.ID,
+				Reason:   reason,
+				Time:     recordTime,
+			}); err != nil {
+				return errors.Wrap(err, "workHandler.AddIncompleteDatapoint")
```

We should probably return the original err from the job execution so that it is captured with the queue record.  If we want this err too we could make it a multi error or something.

---

Yep, see https://github.com/sourcegraph/sourcegraph/pull/45282/commits/0b7d4ca01da77ece54e8676ef5647c5a650ec711